main/acle.md

```diff
@@ -4574,6 +4579,406 @@ two pointers, ignoring the tags.
+
+*   F shares ZT0 with its caller
+
+All other functions have a private-ZA interface.
+
+## Function definitions
+
+<span id="arm_new"></span>
+
+The [keyword attribute](#keyword-attributes) `__arm_new(...)`
+applies to function definitions. It specifies that the function
+creates a [new scope](#new-state-scope) for the given state. `...` is a
+comma-separated list of [state strings](#state-strings). The list
+must be non-empty.
+
+Each piece of state specified by `__arm_new` is zero-initialized.
+
+This attribute does not change a function's binary interface. If the
+function forms part of the object code's ABI, that object code function
+has the same interface as it would have had without `__arm_new`.
+
+A program is [ill-formed](#ill-formed) if a function definition F
+uses `__arm_new` for some state that F also [shares](#shares-state)
+with its caller. For example:
+
+``` c
+  __arm_new("za") void f1() __arm_in("za") { ... }     // Invalid
+  __arm_new("za") void f2() __arm_out("za") { ... }    // Invalid
+  __arm_new("za") void f3() __arm_inout("za") { ... }  // Invalid
+  __arm_new("za") void f4() __arm_inout("zt0") { ... } // OK
+  __arm_new("zt0") void f5() __arm_inout("za") { ... } // OK
+```
+
+## Inline assembly
+
+Some ACLE implementations might support the GNU inline asm extension.
+For implementations that do, suppose that an inline asm occurs in a
+function F that [uses state](#uses-state) S. There are then two
+possibilities:
+
+*   S is not an input to the asm and S is not an output from the asm.
+    The contents of S immediately after executing the asm's instructions
+    are the same as the contents of S immediately before executing the
+    instructions. This is the default assumption.
+
+*   S might be an input to the asm, might be an output from the asm,
+    and might be changed by the asm's instructions. This is indicated by
+    adding S's string to the asm's clobber list. For example, if an asm's
+    clobber list includes `"za"`, ZA might be an input to the asm,
+    an output from the asm, or something that the asm changes.
+
+If instead the inline asm occurs in a function that does not [use
+state](#uses-state) S, the two possibilities are:
+
+*   S is not an input to the asm and S is not an output from the asm.
+    The contents of S immediately after executing the asm's instructions
+    are the same as the contents of S immediately before executing the
+    instructions. This is the default assumption.
+
+*   S is not an input to the asm and S is not an output from the asm.
+    However, the contents of S after executing the asm's instructions might
+    be different from the contents of S before executing the instructions.
+    This is indicated by adding S's string to the asm's clobber list.
+
+    If an asm takes this option for state that is controlled by PSTATE.ZA,
+    the asm itself is responsible for handling the [[AAPCS64]](#AAPCS64)
+    lazy save scheme, as described below.
```

Comment:
I find this sentence a bit confusing. What exactly is it that is described below? It's not really the lazy-save scheme, as this is referenced but not described.

Reply:
I tried saying various alternative things, but I'm not sure they added much, so in the end I just deleted “, as described below”.